introspection_request_handler.go

```diff
@@ -122,16 +123,25 @@ func (f *Fosite) NewIntrospectionRequest(ctx context.Context, r *http.Request, s
 			return &IntrospectionResponse{Active: false}, errors.WithStack(ErrRequestUnauthorized.WithDebug("HTTP Authorization header missing.WithDebug(malformed or credentials used are invalid"))
 		}
 	} else {
-		clientID, clientSecret, ok := r.BasicAuth()
+		id, secret, ok := r.BasicAuth()
 		if !ok {
 			return &IntrospectionResponse{Active: false}, errors.WithStack(ErrRequestUnauthorized.WithDebug("HTTP Authorization header missing.WithDebug(malformed or credentials used are invalid"))
 		}
+		clientID, err := url.QueryUnescape(id)
+		if err != nil {
+			return &IntrospectionResponse{Active: false}, errors.WithStack(ErrRequestUnauthorized.WithDebug("HTTP Authorization header missing.WithDebug(malformed or credentials used are invalid"))
+		}
 
 		client, err := f.Store.GetClient(ctx, clientID)
 		if err != nil {
 			return &IntrospectionResponse{Active: false}, errors.WithStack(ErrRequestUnauthorized.WithDebug("HTTP Authorization header missing.WithDebug(malformed or credentials used are invalid"))
 		}
 
+		clientSecret, err := url.QueryUnescape(secret)
```

Comment:
This should be moved before `GetClient` as it might reduce database strain when passwords are formatted incorrectly.

Reply:
updated